sdk/communication/Azure.Communication.Calling.Server/src/CallClient.cs

```diff
@@ -0,0 +1,516 @@
+        public virtual async Task<Response<CreateCallResponse>> CreateCallAsync(CommunicationIdentifier source, IEnumerable<CommunicationIdentifier> targets, CreateCallOptions options, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(CreateCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNull(source, nameof(source));
+                Argument.AssertNotNullOrEmpty(targets, nameof(targets));
+                Argument.AssertNotNull(options, nameof(options));
+
+                var sourceAlternateIdentity = options.AlternateCallerId == null ? null : new PhoneNumberIdentifierModel(options.AlternateCallerId.PhoneNumber);
+
+                return await RestClient.CreateCallAsync(
+                    targets.Select(t => CommunicationIdentifierSerializer.Serialize(t)),
+                    CommunicationIdentifierSerializer.Serialize(source),
+                    options.CallbackUri?.AbsoluteUri,
+                    options.RequestedModalities,
+                    options.RequestedCallEvents,
+                    sourceAlternateIdentity,
+                    options.Subject,
+                    cancellationToken
+                    ).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// Create an outgoing call from source to target identities.
+        /// <param name="source"> The source identity </param>
+        /// <param name="targets"> The target identities. </param>
+        /// <param name="options"> The call options. </param>
+        /// <param name="cancellationToken"> The cancellation token. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="targets"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="options"/> is null.</exception>
+        public virtual Response<CreateCallResponse> CreateCall(CommunicationIdentifier source, IEnumerable<CommunicationIdentifier> targets, CreateCallOptions options, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(CreateCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNull(source, nameof(source));
+                Argument.AssertNotNullOrEmpty(targets, nameof(targets));
+                Argument.AssertNotNull(options, nameof(options));
```

Comment:
Is the `CreateCallOptions` optional? If so, why do we need to check if it is not null?

Reply:
We check CreateCallOptions because we use its properties and we want to make sure there is no NullReferenceException and let the rest client do the validation for those properties